### PR TITLE
Added tests to resource endpoint

### PR DESCRIPTION
**Pull request by sam-i-am012**
Addresses #22

Has a TODO comment that address #61

### CODE REVIEW

File: src/test/java/org/example/Operations/ResourceOperationsTest.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+        // Create a client with rodsuser status
+        alice = new Wrapper(baseUrl, "alice", "alicepass");
+        alice.authenticate();
+        aliceToken = alice.getAuthToken();
+    }
+
+    @Test
+    @Timeout(5000)
+    public void testCommonOperations() {
```
Comment (korydraughn) [inline src/test/java/org/example/Operations/ResourceOperationsTest.java:58]: What is the motivation behind the `@Timeout`?

File: src/test/java/org/example/Operations/ResourceOperationsTest.java [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+        try {
+            // Create three resources (replication w/ two unixfilesystem resources).
+            res = rods.resourceOperations().create(rodsToken, rescRepl, "replication", null);
+            logger.debug(res.getBody());
+            assertEquals("Creating resource request failed", 200, res.getHttpStatusCode());
+            assertEquals("Creating resource failed",0,
+                    getIrodsResponseStatusCode(res.getBody()));
+
+
```
Comment (korydraughn) [inline src/test/java/org/example/Operations/ResourceOperationsTest.java:73]: Remove one of these empty lines.

File: src/test/java/org/example/Operations/ResourceOperationsTest.java [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+                        getIrodsResponseStatusCode(res.getBody()));
+            }
+
+            // Remove the replication resource.
+            res = rods.resourceOperations().remove(rodsToken, rescRepl);
+            logger.debug(res.getBody());
+            assertEquals("Removing resource request failed", 200, res.getHttpStatusCode());
+            assertEquals("Removing resource failed",0,
+                    getIrodsResponseStatusCode(res.getBody()));
+
```
Comment (korydraughn) [inline src/test/java/org/example/Operations/ResourceOperationsTest.java:209]: Remove empty line.

**Review by korydraughn — COMMENTED** [review]
Do the tests pass?

File: src/test/java/org/example/Operations/ResourceOperationsTest.java (hunk 1 above)
Comment (sam-i-am012) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java:58]: The method that causes the test to fail would take a long time to end, so I had the timeout for that and I forgot to remove it.

File: src/test/java/org/example/Operations/ResourceOperationsTest.java (hunk 2 above)
Comment (sam-i-am012) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java:73]: Done.

File: src/test/java/org/example/Operations/ResourceOperationsTest.java (hunk 3 above)
Comment (sam-i-am012) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java:209]: Done.

File: src/test/java/org/example/Operations/ResourceOperationsTest.java (hunk 1 above)
Comment (korydraughn) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java:58]: So the test is good/passing consistently without that annotation?

Comment (sam-i-am012) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java:58]: That test is still failing, but the tests will still run and complete without the annotation.

Comment (sam-i-am012) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java]: It's about this issue number #61

Comment (korydraughn) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java]: Oh right. I think the issue with that has to do with the fact you're running the tests from the host environment and the HTTP API running in a completely different environment. The HTTP API's python tests assume everything is running local to the iRODS server.

Comment (korydraughn) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java]: Basically, you have a hostname issue.

Comment (sam-i-am012) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java]: Ahh, I see. That makes sense.

Comment (sam-i-am012) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java]: I've marked the test as ignore for now.

**Comment by sam-i-am012** [comment]
All comments have been resolved. Failing test has been marked as ignore.

File: src/test/java/org/example/Operations/ResourceOperationsTest.java [hunk 4]
```diff
@@ -53,6 +54,7 @@ public void setup() {
         aliceToken = alice.getAuthToken();
     }
 
+    @Ignore
```
Comment (korydraughn) [inline src/test/java/org/example/Operations/ResourceOperationsTest.java:57]: I think you can include a message with this.
```java
@Ignore("Remove this annotation once issue #N is resolved")
```
Where `N` is the issue number.

Comment (sam-i-am012) [reply src/test/java/org/example/Operations/ResourceOperationsTest.java]: Done.

**Comment by sam-i-am012** [comment]
All existing comments have been resolved.

**Comment by korydraughn** [comment]
Do the tests pass now?

**Comment by sam-i-am012** [comment]
Yes, they are passing now.

**Comment by korydraughn** [comment]
Please squash to taste, no pounds.

**Comment by korydraughn** [comment]
Let me know once the commits have been squashed.

**Comment by sam-i-am012** [comment]
It's all been squashed.

**Comment by sam-i-am012** [comment]
Done.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by sam-i-am012** [comment]
Done.

**Comment by korydraughn** [comment]
Merging.

**Pull request merged**